sys/include/net/sock/dtls.h

```diff
@@ -468,6 +479,31 @@
  * return 0;
  * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  *
+ * ### Multi-credential handling
+ *
+ * Each sock needs at least one credential tag. Extra credentials can be added
+ * and removed using @ref sock_dtls_add_credential and @ref sock_dtls_remove_credential
+ * respectively (found in `net/sock/dtls/creds.h`).
+ *
+ * #### Pre-shared Keys Cipher Suites
+ * In the case of PSK, a server can optionally indicate a hint to help the server to decide which
+ * PSK Identity to use, using @ref sock_dtls_set_psk_identity_hint. The client application can
+ * decide which credential to use based on the sent hint and/or the session information, by
+ * registering a callback with @ref sock_dtls_set_client_psk_cb. If no callback is registered the
+ * credential is chosen as follows: if a hint is sent by the server, all credentials registered to
+ * the sock are checked for a matching @ref psk_params_t::hint "hint". A credential is select on
+ * matching hint. If no credential matches the hint or no hint is provided, the first PSK credential
```

Comment:
ditto

Reply:
Not sure what exactly needs to change here. I made some adaptions to the doc in 74ea5086deba55b80c490b456d0d84b9f229f5ff. Could you take a look?